lib/streaming-pipeline.js

```diff
@@ -226,13 +226,25 @@ module.exports = class StreamingPipeline extends Writable {
         for (let i = 0; i < inputCount; i++) {
             const unmarshaller = new InputUnmarshaller(transformers[i]);
             unmarshaller.on("error", (err) => {
-                console.error(`error from unmarshaller: ${err.toString()}`);
+                console.error(
+                    `error from ${this._errorOrigin(err)}: ${err.toString()}`
+                );
                 this.emit("error", err);
             });
             this._functionInputs[i] = unmarshaller;
         }
     }
 
+    _errorOrigin(err) {
```

Comment:
is this needed? The error type is already part of the message

Reply:
it's just to make the error message a little less ambiguous, as the prefix would be otherwise "error from unmarshaller: ..." (technically, the error is indeed intercepted from the unmarshaller, but that's an implementation detail)

Reply:
any ambiguity should be addressed when the errors are created, not when they are logged. Unless there is specific context that can be added when logged, but this isn't doing that.

Reply:
let's follow up with another PR